src/group/service.ts

```diff
@@ -32,20 +32,17 @@ export async function addMember(group: Group, user: Principal): Promise<void> {
 export async function replaceMembers(group: Group, users: Principal[]): Promise<void> {
 
   const connection = await database.getConnection();
-  await connection.beginTransaction();
+  const transaction = await connection.transaction();
   try {
-    await connection.query('DELETE FROM group_members WHERE group_id = ?', [group.id]);
+    await transaction.raw('DELETE FROM group_members WHERE group_id = ?', [group.id]);
     for(const user of users) {
-      await connection.query('INSERT INTO group_members SET group_id = ?, user_id = ?', [group.id, user.id]);
+      await transaction.raw('INSERT INTO group_members SET group_id = ?, user_id = ?', [group.id, user.id]);
     }
-    await connection.commit();
+    await transaction.commit();
   } catch (err) {
-    await connection.rollback();
+    await transaction.rollback();
     throw err;
-  } finally {
-    connection.release();
   }
```

Comment:
This is a red flag to me. Most database adapters will give the user a dedicated 'connection' when using `getConnection()`, until it's released back to the pool. Is this not the case with knex?